ts/src/schema/field.ts

```diff
@@ -98,11 +95,65 @@ export function UUIDType(options: FieldOptions): UUIDField {
   return Object.assign(result, options);
 }
 
+export interface IntegerOptions extends FieldOptions {
+  min?: number;
+  max?: number;
+}
+
 export class IntegerField extends BaseField implements Field {
   type: Type = { dbType: DBType.Int };
+  private validators: { (str: number): boolean }[] = [];
+
+  constructor(options?: IntegerOptions) {
+    super();
+    // for legacy callers
+    this.handleOptions(options || { name: "field" });
+  }
+
+  getOptions(): IntegerOptions {
+    return this.getOptions;
+  }
+
+  private handleOptions(options: IntegerOptions) {
+    const params = {
+      min: this.min,
+      max: this.max,
+    };
+
+    for (const k in params) {
+      const v = options[k];
+      if (v !== undefined) {
+        params[k].apply(this, [v]);
+      }
+    }
+  }
+
+  min(l: number): IntegerField {
+    return this.validate((val) => {
+      return val >= l;
+    });
+  }
+
+  max(l: number): IntegerField {
+    return this.validate((val) => val <= l);
+  }
+
+  valid(val: any): boolean {
+    for (const validator of this.validators) {
+      if (!validator(val)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  validate(validator: (str: number) => boolean): IntegerField {
+    this.validators.push(validator);
+    return this;
+  }
 }
 
-export function IntegerType(options: FieldOptions): IntegerField {
+export function IntegerType(options: IntegerOptions): IntegerField {
   let result = new IntegerField();
   return Object.assign(result, options);
```

Comment:
Also not totally sure how this works. Why is it not just `return new IntegerField(options)`?

Reply:
I want the options to be part of the field's properties. used in codegen at https://github.com/lolopinto/ent/blob/main/ts/src/parse_schema/parse.ts#L13